python/pyspark/pandas/strings.py

```diff
@@ -2031,7 +2031,13 @@ def pudf(s: pd.Series) -> pd.Series:
         if expand:
             psdf = psser.to_frame()
             scol = psdf._internal.data_spark_columns[0]
-            spark_columns = [scol[i].alias(str(i)) for i in range(n + 1)]
+
+            if ps.get_option("compute.ansi_mode_support"):
```

Comment:
It seems we can always use the new branch, and remove the configuration read here (which needs one Config RPC)

Reply:
I mean just use
```
                spark_columns = [
                    F.try_element_at(scol, F.lit(i + 1)).alias(str(i)) for i in range(n + 1)
                ]
```
to replace the old one.

since the logic should be the same, I think we don't need the `ps.get_option("compute.ansi_mode_support")` which trigger a Config RPC

Reply:
We might need the config check, "compute.ansi_mode_support" and "spark.sql.ansi.enabled" specifically, for users to switch in different mode combinations. Let me know if you have other concerns.